docs/extensibility/payment-method-integration.md

```diff
@@ -0,0 +1,131 @@
+
+To register an express payment method, you use the `registerExpressPaymentMethod` function from the blocks registry. An example of importing this for use in your JavaScript file is:
+
+_Aliased import_
+```js
+import { registerExpressPaymentMethod } from '@woocommerce/blocks-registry';
+```
+
+_wc global_
+```js
+const { registerExpressPaymentMethod } = wc.wcBlocksRegistry;
+```
+
+The registry function expects a function that will receive an `ExpressPaymentMethodConfig` creator as an argument and is expected to return a valid `ExpressPaymentMethodConfig` instance.  As a very basic example, something like this:
+
+```js
+registerExpressPaymentMethod( ( ExpressPaymentMethodConfig ) => new ExpressPaymentMethodConfig( options ) );
+```
+
+The options you feed the configuration instance should be an object in this shape:
+
+```js
+const options = {
+  name: 'my_payment_method',
+  content: ( <div>A react node</div> ),
+  edit: ( <div> A react node </div> ),
+  canMakePayment: () => true,
+  paymentMethodId: 'new_payment_method',
+}
+```
+
+Here's some more details on the configuration options:
+
+- `name` (required): This should be a unique string (wise to try to pick something unique for your gateway that wouldn't be used by another implementation) that is used as the identifier for the gateway client side. If `paymentMethodId` is not provided, `name` is used for `paymentMethodId` as well.
+- `content` (required): This should be a react node that will output in the express payment method area when the block is rendered in the frontend. It will be cloned in the rendering process. When cloned, this react node will receive props from the payment method interface to checkout that will allow your component to interact with checkout data (more on this later).
+- `edit` (required): This should be a react node that will be output in the express payment method area when the block is rendered in the editor. It will be cloned in the rendering process. When cloned, this react node will receive props from the payment method interface to checkout (but they will contain preview data). 
+- `canMakePayment` (required): This should be a function that returns whether the payment method can make a payment in the current environment. The function _can_ return a promise and will receive [`cartData`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/308e968c700028180cab391f2223eb0a43dd2d4d/assets/js/type-defs/cart.js#L163-L177) as an argument (so the payment method can use that to determine if it can be used). Most payment methods will probably just return `true` from the function, but this does allow payment methods like `ApplePay` which have restrictions on where they work to return `false` and thus be excluded as an option in specific browser environments (or other conditions). Keep in mind this function is only invoked once in the mounting process of the checkout block. 
```

Comment:
Still required? Thought I added a generic one which returns true.

Reply:
You added a generic one that can be reused. Yes I think it should still be required because I _want_ payment methods to consider this. They can just do this now though: `() => true` (it doesn't have to be a promise anymore).

I'm willing to revisit in the future whether we make this optional and default to `() => true`, but it's kind of a critical part of the api - in other words I want to make it hard for payment methods to do things wrong with registering with the api (the assertions will throw helpful errors about what each required option must be and we could improve that messaging over time).